### PR TITLE
Convert to f-strings

### PR DESCRIPTION
**Pull request by nwu63**
## Purpose
I used `flynt` to convert old-style strings to f-strings. In addition, there were a bunch of manual fixes, so it would be helpful of reviewers are careful in reviewing this to make sure I didn't break anything.

## Type of change
What types of change is it?
_Select the appropriate type(s) that describe this PR_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [x] Maintenance update
- [ ] Other (please describe)

## Testing
N/A

## Checklist
_Put an `x` in the boxes that apply._

- [x] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
- [x] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

File: pyoptsparse/pyOpt_variable.py [hunk 1]
```diff
@@ -82,12 +82,7 @@ def __str__(self) -> str:
         if self.type == "d":
             res += "	 "
             res += str(self.name).center(15)
-            res += "{:>25}{:20f} {:14.2e} {:12.2e} \n".format(
```
Comment (marcomangano) [inline pyoptsparse/pyOpt_variable.py:85]: why did you change `self.choices[int(self.value)]` into just `self.value`?

**Review by marcomangano — COMMENTED** [review]
I checked all the changes and they look sound to me, can you confirm that the change below was intentional?

File: pyoptsparse/pyOpt_variable.py (hunk 1 above)
Comment (nwu63) [reply pyoptsparse/pyOpt_variable.py:85]: Good question. It seems like `self.value` is already an integer from above, but at the same time I'm also somewhat certain this stuff is broken since `value` and `self.value` are not the same (in the `__init__`). We do not have any tests for integer DVs I think.

Comment (marcomangano) [reply pyoptsparse/pyOpt_variable.py:85]: Yeah but I think in the first case `self.choices` was a list, and now it's replaced by an integer? Anyway, if we don't have tests and we don't have working examples with this bit of code this is very hard to debug

Comment (nwu63) [reply pyoptsparse/pyOpt_variable.py:85]: Yeah you know what I'll just put this back so this PR is not making any syntax changes. I know we should have tests for this but it's very low on my priority list.

Comment (marcomangano) [reply pyoptsparse/pyOpt_variable.py:85]: yeah I am on the same page, this is not urgent now

**Review by marcomangano — APPROVED** [review]

**Review by sseraj — APPROVED** [review]
